python/ome_tiff_reader.py

```diff
@@ -0,0 +1,147 @@
+#!/usr/bin/env python3
+
+from tifffile import TiffFile
+from apeer_ometiff_library.omexmlClass import OMEXML
+from numcodecs import Zlib
+import zarr
+import numpy as np
+
+import argparse
+from pathlib import Path
+import json
+import urllib
+
+from tile_zarr_base import tile_zarr
+
+DEFAULT_COMPRESSOR = Zlib(level=1)
+DEFAULT_TILE_SIZE = 512
+
+
+def should_be_pyramid(shape):
+    # Generate image pyramid if any dim >= 4096
+    return bool((np.log2(shape) >= 12).any())
+
+
+class OmeTiffReader:
```

Comment:
Maybe add a comment that this only reads tiff and outputs zarr, and does not actually do the tiling/pyramid generation?